### PR TITLE
rework php zero-code page

### PR DESCRIPTION
**Pull request by brettmc**
Addresses some of the feedback raised in https://github.com/open-telemetry/opentelemetry.io/issues/4926

- move example code to page bottom, and note that it can be skipped
- break the page up more into discrete steps to follow
- some editorial improvements
- example of ini-based config

### CODE REVIEW

File: content/en/docs/zero-code/php.md [hunk 1]
```diff
@@ -7,78 +7,23 @@ aliases: [/docs/languages/php/automatic]
...
+Automatic instrumentation with PHP requires:
 
-class DemoClass
-{
-    public function run(): void
-    {
-        echo 'Hello, world';
-    }
-}
+- PHP 8.0 or later
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:14]:
```suggestion
- PHP 8.0 or higher
```

File: content/en/docs/zero-code/php.md [hunk 2]
```diff
@@ -7,78 +7,23 @@ aliases: [/docs/languages/php/automatic]
...
+- the
+  [OpenTelemetry PHP extension](https://github.com/open-telemetry/opentelemetry-php-instrumentation)
+- [composer autoloading](https://getcomposer.org/doc/01-basic-usage.md#autoloading)
+- the [OpenTelemetry SDK](https://packagist.org/packages/open-telemetry/sdk)
+- one or more
+  [instrumentation libraries](/ecosystem/registry/?component=instrumentation&language=php)
+- [configuration](#configuration)
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:21]:
```suggestion
- [OpenTelemetry PHP extension](https://github.com/open-telemetry/opentelemetry-php-instrumentation)
- [Composer autoloading](https://getcomposer.org/doc/01-basic-usage.md#autoloading)
- [OpenTelemetry SDK](https://packagist.org/packages/open-telemetry/sdk)
- One or more
  [instrumentation libraries](/ecosystem/registry/?component=instrumentation&language=php)
- [Configuration](#configuration)
```

File: content/en/docs/zero-code/php.md [hunk 3]
```diff
@@ -7,78 +7,23 @@ aliases: [/docs/languages/php/automatic]
...
-
-In the example above we define `DemoClass`, then register `pre` and `post` hook
-functions on its `run` method. The hook functions will run before and after each
-execution of the `DemoClass::run()` method. The `pre` function starts and
-activates a span, and the `post` function ends it.
-
-If an exception was thrown by `DemoClass::run()`, the `post` function will
-record it, without affecting exception propagation.
+{{% alert title="Important" color="warning" %}}Installing the OpenTelemetry
+extension by itself will not generate traces. {{% /alert %}}
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:24]:
```suggestion
extension by itself does not generate traces. {{% /alert %}}
```

File: content/en/docs/zero-code/php.md [hunk 4]
```diff
@@ -178,10 +123,32 @@ php --ri opentelemetry
    php -m | grep opentelemetry
    ```
 
-## Zero-code configuration for automatic instrumentation
+## Install SDK and instrumentation libraries
+
+Now that the extension is installed, you should install the OpenTelemetry SDK
+and one or more instrumentation libraries.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:129]:
```suggestion
Now that the extension is installed, install the OpenTelemetry SDK
and one or more instrumentation libraries.
```

File: content/en/docs/zero-code/php.md [hunk 5]
```diff
@@ -178,10 +123,32 @@ php --ri opentelemetry
...
+
+Automatic Instrumentation is available for a number commonly used PHP libraries.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:131]:
```suggestion
Automatic instrumentation is available for a number commonly used PHP libraries.
```

File: content/en/docs/zero-code/php.md [hunk 6]
```diff
@@ -178,10 +123,32 @@ php --ri opentelemetry
...
+    open-telemetry/sdk \
+    open-telemetry/opentelemetry-auto-slim \
+    open-telemetry/opentelemetry-auto-psr18
+```
+
+## Configuration
 
 When used in conjunction with the OpenTelemetry SDK, you can use environment
-variables or `php.ini` to configure auto-instrumentation:
+variables or `php.ini` to configure auto-instrumentation.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:149]:
```suggestion
variables or the `php.ini` file to configure auto-instrumentation.
```

File: content/en/docs/zero-code/php.md [hunk 7]
```diff
@@ -178,10 +123,32 @@ php --ri opentelemetry
...
+
+### Via environment
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:151]:
```suggestion
### Environment configuration
```

File: content/en/docs/zero-code/php.md [hunk 8]
```diff
@@ -193,37 +160,106 @@ OTEL_PROPAGATORS=baggage,tracecontext \
 php myapp.php
 ```
 
-## Manual setup for automatic instrumentation
+### Via php.ini
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:163]:
```suggestion
### php.ini configuration
```

File: content/en/docs/zero-code/php.md [hunk 9]
```diff
@@ -193,37 +160,106 @@ OTEL_PROPAGATORS=baggage,tracecontext \
...
+inside the application. In the example above, using Slim Framework and PSR-18
+instrumentation libraries, you should expect to see spans such as:
+
+- a root span representing the HTTP transaction
+- a span for the action that was executed
+- a span for each HTTP transaction that the PSR-18 client sent
+
+It's also worth noting that the PSR-18 client instrumentation will append
+[distributed tracing](/docs/concepts/context-propagation/#propagation) headers
+to outgoing HTTP requests.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:193]:
```suggestion
After all of the above is installed and configured, start your
application as you normally would.

The traces you see exported to the OpenTelemetry Collector depend on the
instrumentation libraries you have installed, and the code path that was taken
inside the application. In the previous example, using Slim Framework and
PSR-18 instrumentation libraries, you should expect to see spans such as:

- A root span representing the HTTP transaction
- A span for the action that was executed
- A span for each HTTP transaction that the PSR-18 client sent

Note that the PSR-18 client instrumentation appends
[distributed tracing](/docs/concepts/context-propagation/#propagation) headers
to outgoing HTTP requests.
```

File: content/en/docs/zero-code/php.md [hunk 10]
```diff
@@ -193,37 +160,106 @@ OTEL_PROPAGATORS=baggage,tracecontext \
...
+instrumentation libraries for your application. {{% /alert %}}
+
+The extension enables registering observer functions (as PHP code) against
+classes and methods, and executing those functions before and after the observed
+method runs.
+
+If there is not an instrumentation library for your framework or application,
+you can write your own. The following example provides some code which we wish
+to auto-instrument, and then illustrates how to use the OpenTelemetry extension
+to trace the execution of that code.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:208]:
```suggestion
The extension enables registering observer functions as PHP code against
classes and methods, and executing those functions before and after the observed
method runs.

If there is not an instrumentation library for your framework or application,
you can write your own. The following example provides some code to be
instrumented, and then illustrates how to use the OpenTelemetry extension
to trace the execution of that code.
```

File: content/en/docs/zero-code/php.md [hunk 11]
```diff
@@ -193,37 +160,106 @@ OTEL_PROPAGATORS=baggage,tracecontext \
...
+/* Run the instrumented code, which will generate a trace */
+$demo = new DemoClass();
+$demo->run();
 ```
 
-## Supported libraries and frameworks
+In the example above we define `DemoClass`, then register `pre` and `post` hook
+functions on its `run` method. The hook functions will run before and after each
+execution of the `DemoClass::run()` method. The `pre` function starts and
+activates a span, and the `post` function ends it.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:259]:
```suggestion
The previous example defines `DemoClass`, then registers `pre` and `post`
hook functions on its `run` method. The hook functions run before and after
each execution of the `DemoClass::run()` method. The `pre` function starts
and activates a span, while the `post` function ends it.
```

File: content/en/docs/zero-code/php.md [hunk 12]
```diff
@@ -193,37 +160,106 @@ OTEL_PROPAGATORS=baggage,tracecontext \
...
 
-Automatic Instrumentation is available for a number of instrumentation libraries
-for commonly used PHP libraries. For the full list, see
-[instrumentation libraries on packagist](https://packagist.org/search/?query=open-telemetry&tags=instrumentation).
+If an exception was thrown by `DemoClass::run()`, the `post` function will
+record it, without affecting exception propagation.
```
Comment (theletterf) [inline content/en/docs/zero-code/php.md:262]:
```suggestion
If `DemoClass::run()` throws an exception, the `post` function
records it without affecting exception propagation.
```

**Review by theletterf — APPROVED** [review]
Good work! Some suggestions added.

**Review by cartermp — APPROVED** [review]
Excellent, thank you!

**Pull request merged**